core/src/main/java/org/apache/iceberg/data/IdentityPartitionConverters.java

```diff
@@ -48,6 +50,13 @@ public static Object convertConstant(Type type, Object value) {
       case FIXED:
         if (value instanceof GenericData.Fixed) {
           return ((GenericData.Fixed) value).bytes();
+        } else if (value instanceof ByteBuffer) {
+          return ((ByteBuffer) value).array();
```

Comment:
Use  
```java
ByteBuffers.toByteArray
```
https://github.com/apache/iceberg/blob/c07f2aabc0a1d02f068ecf1514d2479c0fbdd3b0/api/src/main/java/org/apache/iceberg/util/ByteBuffers.java#L27

A safer option (same suggestion for UUID)